ddtrace/contrib/asgi/middleware.py

```diff
@@ -0,0 +1,113 @@
+from ddtrace import tracer as global_tracer
+from ddtrace.constants import ANALYTICS_SAMPLE_RATE_KEY
+from ddtrace.ext import SpanTypes, http
+from ddtrace.http import store_request_headers, store_response_headers
+from ddtrace.propagation.http import HTTPPropagator
+from ddtrace.settings import config
+from ddtrace.vendor.six.moves import urllib
+
+from ...internal.logger import get_logger
+from .constants import ASGI_SPEC_VERSION, ASGI_VERSION
+from .utils import guarantee_single_callable
+
+log = get_logger(__name__)
+
+config._add("asgi", dict(service_name=config._get_service(default="asgi"), distributed_tracing=True,))
+
+
+def _extract_tags_from_scope(scope):
+    tags = {}
+
+    http_method = scope.get("method")
+    if http_method:
+        tags[http.METHOD] = http_method
+
+    query_string = None
+    if config.asgi.trace_query_string:
+        query_string = scope.get("query_string")
+        if isinstance(query_string, bytes):
+            query_string = query_string.decode()
+        tags[http.QUERY_STRING] = query_string
+
+    server = scope.get("server")
+    if server and len(server) == 2:
+        port = server[1]
+        server_host = server[0] + (":" + str(port) if port != 80 else "")
+        full_path = scope.get("root_path", "") + scope.get("path", "")
+        http_url = scope.get("scheme", "http") + "://" + server_host + full_path
+        if query_string:
+            http_url = http_url + ("?" + urllib.parse.unquote(query_string))
```

Comment:
Hmm so we'd be including the query string here twice right? Under `tags[http.QUERY_STRING]` and `tags[http.URL]`. I don't think we typically include it with the url so we can exclude it here.